File: elixir/gigasecond/lib/gigasecond.ex
defmodule Gigasecond do
  @doc """
  Calculate a date one billion seconds after an input date.
  """
  @spec from({{pos_integer, pos_integer, pos_integer}, {pos_integer, pos_integer, pos_integer}}) ::
          :calendar.datetime()

  def from({{year, month, day}, {hours, minutes, seconds}}) do
    {{year, month, day}, {hours, minutes, seconds}}
    |> NaiveDateTime.from_erl!()
    |> NaiveDateTime.add(1_000_000_000, :second)
    |> NaiveDateTime.to_erl()
  end
end

File: clojure/gigasecond/src/gigasecond.clj
(ns gigasecond
  (:use [clojure.core]
        [clj-time.core :as t]))

(defn from [y m d]
  (let [initial (t/date-time y m d)
        final (t/plus initial (t/seconds 1000000000))]
    ((juxt t/year t/month t/day) final)))
